roles/test-utils/mining-device/src/main.rs

```diff
@@ -614,33 +613,40 @@ impl Miner {
 
 // returns hashrate based on how fast the device hashes over the given duration
 fn measure_hashrate(duration_secs: u64) -> f64 {
-    let mut share = generate_random_80_byte_array();
+    let mut rng = thread_rng();
+    let prev_hash: [u8; 32] = generate_random_32_byte_array().to_vec().try_into().unwrap();
+    let prev_hash = Hash::from_inner(prev_hash);
+    let merkle_root: [u8; 32] = generate_random_32_byte_array().to_vec().try_into().unwrap();
+    let merkle_root = Hash::from_inner(merkle_root);
+    let header = BlockHeader {
+        version: rng.gen(),
+        prev_blockhash: BlockHash::from_hash(prev_hash),
+        merkle_root,
+        time: std::time::SystemTime::now()
+            .duration_since(std::time::SystemTime::UNIX_EPOCH - std::time::Duration::from_secs(60))
+            .unwrap()
+            .as_secs() as u32,
+        bits: rng.gen(),
+        nonce: 0,
+    };
     let start_time = Instant::now();
     let mut hashes: u64 = 0;
     let duration = Duration::from_secs(duration_secs);
+    let mut miner = Miner::new(0);
+    miner.new_target(vec![0_u8; 32]);
```

Comment:
why are we setting the target as 0?

doesn't this make is basically impossible to find a share?